<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../../../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../../../jacoco-resources/report.gif" type="image/gif"/><title>CreatureTest.java</title><link rel="stylesheet" href="../../../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../../../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../../../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../../../index.html" class="el_report">AllUiTests (May 19, 2018 11:46:29 AM)</a> &gt; <a href="../../index.html" class="el_group">JoshHack</a> &gt; <a href="../index.html" class="el_bundle">src</a> &gt; <a href="index.source.html" class="el_package">game</a> &gt; <span class="el_source">CreatureTest.java</span></div><h1>CreatureTest.java</h1><pre class="source lang-java linenums">package game;

import java.awt.Color;
import java.util.List;
import org.junit.*;
import static org.junit.Assert.*;

/**
 * The class &lt;code&gt;CreatureTest&lt;/code&gt; contains tests for the class &lt;code&gt;{@link Creature}&lt;/code&gt;.
 *
 * @generatedBy CodePro at 5/19/18 11:03 AM
 * @author Collin
 * @version $Revision: 1.0 $
 */
<span class="nc" id="L15">public class CreatureTest {</span>
	/**
	 * Run the Creature(World,char,Color,String,int,int,int) constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCreature_1()
		throws Exception {
<span class="nc" id="L26">		World world = WorldFactory.createWorld();</span>
<span class="nc" id="L27">		char glyph = '';</span>
<span class="nc" id="L28">		Color color = new Color(1);</span>
<span class="nc" id="L29">		String name = &quot;&quot;;</span>
<span class="nc" id="L30">		int maxHp = 1;</span>
<span class="nc" id="L31">		int attack = 1;</span>
<span class="nc" id="L32">		int defense = 1;</span>

<span class="nc" id="L34">		Creature result = new Creature(world, glyph, color, name, maxHp, attack, defense);</span>

		// add additional test code here
<span class="nc" id="L37">		assertNotNull(result);</span>
<span class="nc" id="L38">		assertEquals(&quot;&quot;, result.name());</span>
<span class="nc" id="L39">		assertEquals(1, result.level());</span>
<span class="nc" id="L40">		assertEquals(&quot;  level:1  attack:1  defense:1  hp:1&quot;, result.details());</span>
<span class="nc" id="L41">		assertEquals('', result.glyph());</span>
<span class="nc" id="L42">		assertEquals(null, result.weapon());</span>
<span class="nc" id="L43">		assertEquals(null, result.armor());</span>
<span class="nc" id="L44">		assertEquals(5, result.maxMana());</span>
<span class="nc" id="L45">		assertEquals(0, result.xp());</span>
<span class="nc" id="L46">		assertEquals(666, result.food());</span>
<span class="nc" id="L47">		assertEquals(9, result.visionRadius());</span>
<span class="nc" id="L48">		assertEquals(1, result.maxHp());</span>
<span class="nc" id="L49">		assertEquals(1, result.hp());</span>
<span class="nc" id="L50">		assertEquals(1000, result.maxFood());</span>
<span class="nc" id="L51">		assertEquals(1, result.defenseValue());</span>
<span class="nc" id="L52">		assertEquals(5, result.mana());</span>
<span class="nc" id="L53">		assertEquals(null, result.causeOfDeath());</span>
<span class="nc" id="L54">		assertEquals(1, result.attackValue());</span>
<span class="nc" id="L55">		assertEquals(false, result.isPlayer());</span>
<span class="nc" id="L56">	}</span>

	/**
	 * Run the Item armor() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testArmor_1()
		throws Exception {
<span class="nc" id="L68">		Creature fixture = CreatureFactory.createCreature6();</span>

<span class="nc" id="L70">		Item result = fixture.armor();</span>

		// add additional test code here
<span class="nc" id="L73">		assertEquals(null, result);</span>
<span class="nc" id="L74">	}</span>

	/**
	 * Run the int attackValue() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testAttackValue_1()
		throws Exception {
<span class="nc" id="L86">		Creature fixture = CreatureFactory.createCreature();</span>

<span class="nc" id="L88">		int result = fixture.attackValue();</span>

		// add additional test code here
<span class="nc" id="L91">		assertEquals(0, result);</span>
<span class="nc" id="L92">	}</span>

	/**
	 * Run the int attackValue() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testAttackValue_2()
		throws Exception {
<span class="nc" id="L104">		Creature fixture = CreatureFactory.createCreature2();</span>

<span class="nc" id="L106">		int result = fixture.attackValue();</span>

		// add additional test code here
<span class="nc" id="L109">		assertEquals(Integer.MIN_VALUE, result);</span>
<span class="nc" id="L110">	}</span>

	/**
	 * Run the boolean canEnter(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCanEnter_1()
		throws Exception {
<span class="nc" id="L122">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L123">		int wx = 1;</span>
<span class="nc" id="L124">		int wy = 1;</span>
<span class="nc" id="L125">		int wz = 1;</span>

<span class="nc" id="L127">		boolean result = fixture.canEnter(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canEnter(Creature.java:280)
<span class="nc" id="L133">		assertTrue(result);</span>
<span class="nc" id="L134">	}</span>

	/**
	 * Run the boolean canEnter(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCanEnter_2()
		throws Exception {
<span class="nc" id="L146">		Creature fixture = CreatureFactory.createCreature7();</span>
<span class="nc" id="L147">		int wx = 1;</span>
<span class="nc" id="L148">		int wy = 1;</span>
<span class="nc" id="L149">		int wz = 1;</span>

<span class="nc" id="L151">		boolean result = fixture.canEnter(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canEnter(Creature.java:280)
<span class="nc" id="L157">		assertTrue(result);</span>
<span class="nc" id="L158">	}</span>

	/**
	 * Run the boolean canEnter(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCanEnter_3()
		throws Exception {
<span class="nc" id="L170">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L171">		int wx = 1;</span>
<span class="nc" id="L172">		int wy = 1;</span>
<span class="nc" id="L173">		int wz = 1;</span>

<span class="nc" id="L175">		boolean result = fixture.canEnter(wx, wy, wz);</span>

		// add additional test code here
<span class="nc" id="L178">		assertEquals(false, result);</span>
<span class="nc" id="L179">	}</span>

	/**
	 * Run the boolean canSee(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCanSee_1()
		throws Exception {
<span class="nc" id="L191">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L192">		int wx = 1;</span>
<span class="nc" id="L193">		int wy = 1;</span>
<span class="nc" id="L194">		int wz = 1;</span>

<span class="nc" id="L196">		boolean result = fixture.canSee(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
<span class="nc" id="L202">		assertTrue(result);</span>
<span class="nc" id="L203">	}</span>

	/**
	 * Run the boolean canSee(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCanSee_2()
		throws Exception {
<span class="nc" id="L215">		Creature fixture = CreatureFactory.createCreature4();</span>
<span class="nc" id="L216">		int wx = 1;</span>
<span class="nc" id="L217">		int wy = 1;</span>
<span class="nc" id="L218">		int wz = 1;</span>

<span class="nc" id="L220">		boolean result = fixture.canSee(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
<span class="nc" id="L226">		assertTrue(result);</span>
<span class="nc" id="L227">	}</span>

	/**
	 * Run the boolean canSee(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCanSee_3()
		throws Exception {
<span class="nc" id="L239">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L240">		int wx = 1;</span>
<span class="nc" id="L241">		int wy = 1;</span>
<span class="nc" id="L242">		int wz = 1;</span>

<span class="nc" id="L244">		boolean result = fixture.canSee(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
<span class="nc" id="L250">		assertTrue(result);</span>
<span class="nc" id="L251">	}</span>

	/**
	 * Run the String causeOfDeath() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCauseOfDeath_1()
		throws Exception {
<span class="nc" id="L263">		Creature fixture = CreatureFactory.createCreature2();</span>

<span class="nc" id="L265">		String result = fixture.causeOfDeath();</span>

		// add additional test code here
<span class="nc" id="L268">		assertEquals(null, result);</span>
<span class="nc" id="L269">	}</span>

	/**
	 * Run the Color color() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testColor_1()
		throws Exception {
<span class="nc" id="L281">		Creature fixture = CreatureFactory.createCreature2();</span>

<span class="nc" id="L283">		Color result = fixture.color();</span>

		// add additional test code here
<span class="nc" id="L286">		assertNotNull(result);</span>
<span class="nc" id="L287">		assertEquals(&quot;java.awt.Color[r=192,g=192,b=192]&quot;, result.toString());</span>
<span class="nc" id="L288">		assertEquals(192, result.getGreen());</span>
<span class="nc" id="L289">		assertEquals(192, result.getRed());</span>
<span class="nc" id="L290">		assertEquals(192, result.getBlue());</span>
<span class="nc" id="L291">		assertEquals(255, result.getAlpha());</span>
<span class="nc" id="L292">		assertEquals(-4144960, result.getRGB());</span>
<span class="nc" id="L293">		assertEquals(1, result.getTransparency());</span>
<span class="nc" id="L294">	}</span>

	/**
	 * Run the Creature creature(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCreature_2()
		throws Exception {
<span class="nc" id="L306">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L307">		int wx = 1;</span>
<span class="nc" id="L308">		int wy = 1;</span>
<span class="nc" id="L309">		int wz = 1;</span>

<span class="nc" id="L311">		Creature result = fixture.creature(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
<span class="nc" id="L318">		assertNotNull(result);</span>
<span class="nc" id="L319">	}</span>

	/**
	 * Run the Creature creature(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testCreature_3()
		throws Exception {
<span class="nc" id="L331">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L332">		int wx = 1;</span>
<span class="nc" id="L333">		int wy = 1;</span>
<span class="nc" id="L334">		int wz = 1;</span>

<span class="nc" id="L336">		Creature result = fixture.creature(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
<span class="nc" id="L343">		assertNotNull(result);</span>
<span class="nc" id="L344">	}</span>

	/**
	 * Run the int defenseValue() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDefenseValue_1()
		throws Exception {
<span class="nc" id="L356">		Creature fixture = CreatureFactory.createCreature4();</span>

<span class="nc" id="L358">		int result = fixture.defenseValue();</span>

		// add additional test code here
<span class="nc" id="L361">		assertEquals(Integer.MAX_VALUE, result);</span>
<span class="nc" id="L362">	}</span>

	/**
	 * Run the int defenseValue() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDefenseValue_2()
		throws Exception {
<span class="nc" id="L374">		Creature fixture = CreatureFactory.createCreature5();</span>

<span class="nc" id="L376">		int result = fixture.defenseValue();</span>

		// add additional test code here
<span class="nc" id="L379">		assertEquals(1, result);</span>
<span class="nc" id="L380">	}</span>

	/**
	 * Run the String details() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDetails_1()
		throws Exception {
<span class="nc" id="L392">		Creature fixture = CreatureFactory.createCreature3();</span>

<span class="nc" id="L394">		String result = fixture.details();</span>

		// add additional test code here
<span class="nc" id="L397">		assertEquals(&quot;  level:1  attack:7  defense:7  hp:7&quot;, result);</span>
<span class="nc" id="L398">	}</span>

	/**
	 * Run the void doAction(String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_1()
		throws Exception {
<span class="nc" id="L410">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L411">		String message = &quot;&quot;;</span>

<span class="nc" id="L413">		fixture.doAction(message);</span>

		// add additional test code here
<span class="nc" id="L416">	}</span>

	/**
	 * Run the void doAction(String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_2()
		throws Exception {
<span class="nc" id="L428">		Creature fixture = CreatureFactory.createCreature4();</span>
<span class="nc" id="L429">		String message = &quot;&quot;;</span>

<span class="nc" id="L431">		fixture.doAction(message);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.getCreaturesWhoSeeMe(Creature.java:319)
		//       at game.Creature.doAction(Creature.java:288)
<span class="nc" id="L438">	}</span>

	/**
	 * Run the void doAction(String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_3()
		throws Exception {
<span class="nc" id="L450">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L451">		String message = &quot;&quot;;</span>

<span class="nc" id="L453">		fixture.doAction(message);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.getCreaturesWhoSeeMe(Creature.java:319)
		//       at game.Creature.doAction(Creature.java:288)
<span class="nc" id="L460">	}</span>

	/**
	 * Run the void doAction(Item,String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_4()
		throws Exception {
<span class="nc" id="L472">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L473">		Item item = ItemFactory.createItem();</span>
<span class="nc" id="L474">		String message = &quot;&quot;;</span>

<span class="nc" id="L476">		fixture.doAction(item, message);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.getCreaturesWhoSeeMe(Creature.java:319)
		//       at game.Creature.doAction(Creature.java:301)
<span class="nc" id="L483">	}</span>

	/**
	 * Run the void doAction(Item,String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_5()
		throws Exception {
<span class="nc" id="L495">		Creature fixture = CreatureFactory.createCreature7();</span>
<span class="nc" id="L496">		Item item = ItemFactory.createItem2();</span>
<span class="nc" id="L497">		String message = &quot;&quot;;</span>

<span class="nc" id="L499">		fixture.doAction(item, message);</span>

		// add additional test code here
<span class="nc" id="L502">	}</span>

	/**
	 * Run the void doAction(Item,String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_6()
		throws Exception {
<span class="nc" id="L514">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L515">		Item item = ItemFactory.createItem3();</span>
<span class="nc" id="L516">		String message = &quot;&quot;;</span>

<span class="nc" id="L518">		fixture.doAction(item, message);</span>

		// add additional test code here
<span class="nc" id="L521">	}</span>

	/**
	 * Run the void doAction(Item,String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDoAction_7()
		throws Exception {
<span class="nc" id="L533">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L534">		Item item = ItemFactory.createItem4();</span>
<span class="nc" id="L535">		String message = &quot;&quot;;</span>

<span class="nc" id="L537">		fixture.doAction(item, message);</span>

		// add additional test code here
<span class="nc" id="L540">	}</span>

	/**
	 * Run the void drop(Item) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testDrop_1()
		throws Exception {
<span class="nc" id="L552">		Creature fixture = CreatureFactory.createCreature4();</span>
<span class="nc" id="L553">		Item item = ItemFactory.createItem();</span>

<span class="nc" id="L555">		fixture.drop(item);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.drop(Creature.java:379)
<span class="nc" id="L561">	}</span>

	/**
	 * Run the List&lt;Effect&gt; effects() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testEffects_1()
		throws Exception {
<span class="nc" id="L573">		Creature fixture = CreatureFactory.createCreature3();</span>

<span class="nc" id="L575">		List&lt;Effect&gt; result = fixture.effects();</span>

		// add additional test code here
<span class="nc" id="L578">		assertNotNull(result);</span>
<span class="nc" id="L579">		assertEquals(0, result.size());</span>
<span class="nc" id="L580">	}</span>

	/**
	 * Run the int food() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testFood_1()
		throws Exception {
<span class="nc" id="L592">		Creature fixture = CreatureFactory.createCreature4();</span>

<span class="nc" id="L594">		int result = fixture.food();</span>

		// add additional test code here
<span class="nc" id="L597">		assertEquals(666, result);</span>
<span class="nc" id="L598">	}</span>

	/**
	 * Run the void gainXp(Creature) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testGainXp_1()
		throws Exception {
<span class="nc" id="L610">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L611">		Creature other = CreatureFactory.createCreature2();</span>

<span class="nc" id="L613">		fixture.gainXp(other);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.notify(Creature.java:284)
		//       at game.Creature.modifyXp(Creature.java:74)
		//       at game.Creature.gainXp(Creature.java:203)
<span class="nc" id="L621">	}</span>

	/**
	 * Run the void gainXp(Creature) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testGainXp_2()
		throws Exception {
<span class="nc" id="L633">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L634">		Creature other = CreatureFactory.createCreature4();</span>

<span class="nc" id="L636">		fixture.gainXp(other);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.notify(Creature.java:284)
		//       at game.Creature.modifyXp(Creature.java:74)
		//       at game.Creature.gainXp(Creature.java:203)
<span class="nc" id="L644">	}</span>

	/**
	 * Run the char glyph() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testGlyph_1()
		throws Exception {
<span class="nc" id="L656">		Creature fixture = CreatureFactory.createCreature();</span>

<span class="nc" id="L658">		char result = fixture.glyph();</span>

		// add additional test code here
<span class="nc" id="L661">		assertEquals(' ', result);</span>
<span class="nc" id="L662">	}</span>

	/**
	 * Run the int hp() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testHp_1()
		throws Exception {
<span class="nc" id="L674">		Creature fixture = CreatureFactory.createCreature6();</span>

<span class="nc" id="L676">		int result = fixture.hp();</span>

		// add additional test code here
<span class="nc" id="L679">		assertEquals(1, result);</span>
<span class="nc" id="L680">	}</span>

	/**
	 * Run the Inventory inventory() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testInventory_1()
		throws Exception {
<span class="nc" id="L692">		Creature fixture = CreatureFactory.createCreature2();</span>

<span class="nc" id="L694">		Inventory result = fixture.inventory();</span>

		// add additional test code here
<span class="nc" id="L697">		assertNotNull(result);</span>
<span class="nc" id="L698">		assertEquals(false, result.isFull());</span>
<span class="nc" id="L699">	}</span>

	/**
	 * Run the boolean isPlayer() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testIsPlayer_1()
		throws Exception {
<span class="nc" id="L711">		Creature fixture = CreatureFactory.createCreature6();</span>

<span class="nc" id="L713">		boolean result = fixture.isPlayer();</span>

		// add additional test code here
<span class="nc" id="L716">		assertEquals(false, result);</span>
<span class="nc" id="L717">	}</span>

	/**
	 * Run the boolean isPlayer() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testIsPlayer_2()
		throws Exception {
<span class="nc" id="L729">		Creature fixture = CreatureFactory.createCreature7();</span>

<span class="nc" id="L731">		boolean result = fixture.isPlayer();</span>

		// add additional test code here
<span class="nc" id="L734">		assertEquals(false, result);</span>
<span class="nc" id="L735">	}</span>

	/**
	 * Run the Item item(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testItem_1()
		throws Exception {
<span class="nc" id="L747">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L748">		int wx = 1;</span>
<span class="nc" id="L749">		int wy = 1;</span>
<span class="nc" id="L750">		int wz = 1;</span>

<span class="nc" id="L752">		Item result = fixture.item(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.item(Creature.java:485)
<span class="nc" id="L759">		assertNotNull(result);</span>
<span class="nc" id="L760">	}</span>

	/**
	 * Run the Item item(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testItem_2()
		throws Exception {
<span class="nc" id="L772">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L773">		int wx = 1;</span>
<span class="nc" id="L774">		int wy = 1;</span>
<span class="nc" id="L775">		int wz = 1;</span>

<span class="nc" id="L777">		Item result = fixture.item(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.item(Creature.java:485)
<span class="nc" id="L784">		assertNotNull(result);</span>
<span class="nc" id="L785">	}</span>

	/**
	 * Run the int level() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testLevel_1()
		throws Exception {
<span class="nc" id="L797">		Creature fixture = CreatureFactory.createCreature();</span>

<span class="nc" id="L799">		int result = fixture.level();</span>

		// add additional test code here
<span class="nc" id="L802">		assertEquals(1, result);</span>
<span class="nc" id="L803">	}</span>

	/**
	 * Run the int mana() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMana_1()
		throws Exception {
<span class="nc" id="L815">		Creature fixture = CreatureFactory.createCreature6();</span>

<span class="nc" id="L817">		int result = fixture.mana();</span>

		// add additional test code here
<span class="nc" id="L820">		assertEquals(5, result);</span>
<span class="nc" id="L821">	}</span>

	/**
	 * Run the int maxFood() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMaxFood_1()
		throws Exception {
<span class="nc" id="L833">		Creature fixture = CreatureFactory.createCreature3();</span>

<span class="nc" id="L835">		int result = fixture.maxFood();</span>

		// add additional test code here
<span class="nc" id="L838">		assertEquals(1000, result);</span>
<span class="nc" id="L839">	}</span>

	/**
	 * Run the int maxHp() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMaxHp_1()
		throws Exception {
<span class="nc" id="L851">		Creature fixture = CreatureFactory.createCreature4();</span>

<span class="nc" id="L853">		int result = fixture.maxHp();</span>

		// add additional test code here
<span class="nc" id="L856">		assertEquals(Integer.MAX_VALUE, result);</span>
<span class="nc" id="L857">	}</span>

	/**
	 * Run the int maxMana() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMaxMana_1()
		throws Exception {
<span class="nc" id="L869">		Creature fixture = CreatureFactory.createCreature4();</span>

<span class="nc" id="L871">		int result = fixture.maxMana();</span>

		// add additional test code here
<span class="nc" id="L874">		assertEquals(5, result);</span>
<span class="nc" id="L875">	}</span>

	/**
	 * Run the void meleeAttack(Creature) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMeleeAttack_1()
		throws Exception {
<span class="nc" id="L887">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L888">		Creature other = CreatureFactory.createCreature7();</span>

<span class="nc" id="L890">		fixture.meleeAttack(other);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.getCreaturesWhoSeeMe(Creature.java:319)
		//       at game.Creature.doAction(Creature.java:288)
		//       at game.Creature.commonAttack(Creature.java:188)
		//       at game.Creature.meleeAttack(Creature.java:163)
<span class="nc" id="L899">	}</span>

	/**
	 * Run the void modifyAttackValue(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyAttackValue_1()
		throws Exception {
<span class="nc" id="L911">		Creature fixture = CreatureFactory.createCreature7();</span>
<span class="nc" id="L912">		int value = 1;</span>

<span class="nc" id="L914">		fixture.modifyAttackValue(value);</span>

		// add additional test code here
<span class="nc" id="L917">	}</span>

	/**
	 * Run the void modifyDefenseValue(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyDefenseValue_1()
		throws Exception {
<span class="nc" id="L929">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L930">		int value = 1;</span>

<span class="nc" id="L932">		fixture.modifyDefenseValue(value);</span>

		// add additional test code here
<span class="nc" id="L935">	}</span>

	/**
	 * Run the void modifyDetectCreatures(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyDetectCreatures_1()
		throws Exception {
<span class="nc" id="L947">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L948">		int amount = 1;</span>

<span class="nc" id="L950">		fixture.modifyDetectCreatures(amount);</span>

		// add additional test code here
<span class="nc" id="L953">	}</span>

	/**
	 * Run the void modifyFood(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyFood_1()
		throws Exception {
<span class="nc" id="L965">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L966">		int amount = 1;</span>

<span class="nc" id="L968">		fixture.modifyFood(amount);</span>

		// add additional test code here
<span class="nc" id="L971">	}</span>

	/**
	 * Run the void modifyMana(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyMana_1()
		throws Exception {
<span class="nc" id="L983">		Creature fixture = CreatureFactory.createCreature7();</span>
<span class="nc" id="L984">		int amount = 1;</span>

<span class="nc" id="L986">		fixture.modifyMana(amount);</span>

		// add additional test code here
<span class="nc" id="L989">	}</span>

	/**
	 * Run the void modifyMaxHp(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyMaxHp_1()
		throws Exception {
<span class="nc" id="L1001">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L1002">		int amount = 1;</span>

<span class="nc" id="L1004">		fixture.modifyMaxHp(amount);</span>

		// add additional test code here
<span class="nc" id="L1007">	}</span>

	/**
	 * Run the void modifyMaxMana(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyMaxMana_1()
		throws Exception {
<span class="nc" id="L1019">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L1020">		int amount = 1;</span>

<span class="nc" id="L1022">		fixture.modifyMaxMana(amount);</span>

		// add additional test code here
<span class="nc" id="L1025">	}</span>

	/**
	 * Run the void modifyRegenHpPer1000(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyRegenHpPer1000_1()
		throws Exception {
<span class="nc" id="L1037">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L1038">		int amount = 1;</span>

<span class="nc" id="L1040">		fixture.modifyRegenHpPer1000(amount);</span>

		// add additional test code here
<span class="nc" id="L1043">	}</span>

	/**
	 * Run the void modifyRegenManaPer1000(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyRegenManaPer1000_1()
		throws Exception {
<span class="nc" id="L1055">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L1056">		int amount = 1;</span>

<span class="nc" id="L1058">		fixture.modifyRegenManaPer1000(amount);</span>

		// add additional test code here
<span class="nc" id="L1061">	}</span>

	/**
	 * Run the void modifyVisionRadius(int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testModifyVisionRadius_1()
		throws Exception {
<span class="nc" id="L1073">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L1074">		int value = 1;</span>

<span class="nc" id="L1076">		fixture.modifyVisionRadius(value);</span>

		// add additional test code here
<span class="nc" id="L1079">	}</span>

	/**
	 * Run the void moveBy(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMoveBy_1()
		throws Exception {
<span class="nc" id="L1091">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L1092">		int mx = 0;</span>
<span class="nc" id="L1093">		int my = 0;</span>
<span class="nc" id="L1094">		int mz = 1;</span>

<span class="nc" id="L1096">		fixture.moveBy(mx, my, mz);</span>

		// add additional test code here
<span class="nc" id="L1099">	}</span>

	/**
	 * Run the void moveBy(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMoveBy_2()
		throws Exception {
<span class="nc" id="L1111">		Creature fixture = CreatureFactory.createCreature4();</span>
<span class="nc" id="L1112">		int mx = 1;</span>
<span class="nc" id="L1113">		int my = 1;</span>
<span class="nc" id="L1114">		int mz = 2;</span>

<span class="nc" id="L1116">		fixture.moveBy(mx, my, mz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.moveBy(Creature.java:134)
<span class="nc" id="L1122">	}</span>

	/**
	 * Run the void moveBy(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testMoveBy_3()
		throws Exception {
<span class="nc" id="L1134">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L1135">		int mx = 0;</span>
<span class="nc" id="L1136">		int my = 1;</span>
<span class="nc" id="L1137">		int mz = -1;</span>

<span class="nc" id="L1139">		fixture.moveBy(mx, my, mz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.moveBy(Creature.java:134)
<span class="nc" id="L1145">	}</span>

	/**
	 * Run the String name() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testName_1()
		throws Exception {
<span class="nc" id="L1157">		Creature fixture = CreatureFactory.createCreature();</span>

<span class="nc" id="L1159">		String result = fixture.name();</span>

		// add additional test code here
<span class="nc" id="L1162">		assertEquals(&quot;&quot;, result);</span>
<span class="nc" id="L1163">	}</span>

	/**
	 * Run the String nameOf(Item) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testNameOf_1()
		throws Exception {
<span class="nc" id="L1175">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L1176">		Item item = ItemFactory.createItem2();</span>

<span class="nc" id="L1178">		String result = fixture.nameOf(item);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.nameOf(Creature.java:543)
<span class="nc" id="L1184">		assertNotNull(result);</span>
<span class="nc" id="L1185">	}</span>

	/**
	 * Run the void notify(String,Object[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testNotify_1()
		throws Exception {
<span class="nc" id="L1197">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L1198">		String message = &quot;&quot;;</span>

<span class="nc" id="L1200">		fixture.notify(message);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.notify(Creature.java:284)
<span class="nc" id="L1206">	}</span>

	/**
	 * Run the Tile realTile(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testRealTile_1()
		throws Exception {
<span class="nc" id="L1218">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L1219">		int wx = 1;</span>
<span class="nc" id="L1220">		int wy = 1;</span>
<span class="nc" id="L1221">		int wz = 1;</span>

<span class="nc" id="L1223">		Tile result = fixture.realTile(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
<span class="nc" id="L1229">		assertNotNull(result);</span>
<span class="nc" id="L1230">	}</span>

	/**
	 * Run the void setCreatureAi(CreatureAi) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testSetCreatureAi_1()
		throws Exception {
<span class="nc" id="L1242">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L1243">		CreatureAi ai = CreatureAiFactory.createCreatureAi();</span>

<span class="nc" id="L1245">		fixture.setCreatureAi(ai);</span>

		// add additional test code here
<span class="nc" id="L1248">	}</span>

	/**
	 * Run the void summon(Creature) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testSummon_1()
		throws Exception {
<span class="nc" id="L1260">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L1261">		Creature other = CreatureFactory.createCreature7();</span>

<span class="nc" id="L1263">		fixture.summon(other);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.summon(Creature.java:521)
<span class="nc" id="L1269">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_1()
		throws Exception {
<span class="nc" id="L1281">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L1282">		Item item = ItemFactory.createItem2();</span>
<span class="nc" id="L1283">		int wx = 1;</span>
<span class="nc" id="L1284">		int wy = 1;</span>
<span class="nc" id="L1285">		int wz = 1;</span>

<span class="nc" id="L1287">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
		//       at game.Creature.throwItem(Creature.java:507)
<span class="nc" id="L1295">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_2()
		throws Exception {
<span class="nc" id="L1307">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L1308">		Item item = ItemFactory.createItem3();</span>
<span class="nc" id="L1309">		int wx = 1;</span>
<span class="nc" id="L1310">		int wy = 1;</span>
<span class="nc" id="L1311">		int wz = 1;</span>

<span class="nc" id="L1313">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
		//       at game.Creature.throwItem(Creature.java:507)
<span class="nc" id="L1321">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_3()
		throws Exception {
<span class="nc" id="L1333">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L1334">		Item item = ItemFactory.createItem4();</span>
<span class="nc" id="L1335">		int wx = 1;</span>
<span class="nc" id="L1336">		int wy = 1;</span>
<span class="nc" id="L1337">		int wz = 1;</span>

<span class="nc" id="L1339">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
		//       at game.Creature.throwItem(Creature.java:507)
<span class="nc" id="L1347">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_4()
		throws Exception {
<span class="nc" id="L1359">		Creature fixture = CreatureFactory.createCreature4();</span>
<span class="nc" id="L1360">		Item item = ItemFactory.createItem();</span>
<span class="nc" id="L1361">		int wx = 1;</span>
<span class="nc" id="L1362">		int wy = 1;</span>
<span class="nc" id="L1363">		int wz = 1;</span>

<span class="nc" id="L1365">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
		//       at game.Creature.throwItem(Creature.java:499)
<span class="nc" id="L1372">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_5()
		throws Exception {
<span class="nc" id="L1384">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L1385">		Item item = ItemFactory.createItem2();</span>
<span class="nc" id="L1386">		int wx = 1;</span>
<span class="nc" id="L1387">		int wy = 1;</span>
<span class="nc" id="L1388">		int wz = 1;</span>

<span class="nc" id="L1390">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
		//       at game.Creature.throwItem(Creature.java:499)
<span class="nc" id="L1397">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_6()
		throws Exception {
<span class="nc" id="L1409">		Creature fixture = CreatureFactory.createCreature6();</span>
<span class="nc" id="L1410">		Item item = ItemFactory.createItem3();</span>
<span class="nc" id="L1411">		int wx = 1;</span>
<span class="nc" id="L1412">		int wy = 1;</span>
<span class="nc" id="L1413">		int wz = 1;</span>

<span class="nc" id="L1415">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
		//       at game.Creature.throwItem(Creature.java:499)
<span class="nc" id="L1422">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_7()
		throws Exception {
<span class="nc" id="L1434">		Creature fixture = CreatureFactory.createCreature7();</span>
<span class="nc" id="L1435">		Item item = ItemFactory.createItem4();</span>
<span class="nc" id="L1436">		int wx = 1;</span>
<span class="nc" id="L1437">		int wy = 1;</span>
<span class="nc" id="L1438">		int wz = 1;</span>

<span class="nc" id="L1440">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
		//       at game.Creature.throwItem(Creature.java:499)
<span class="nc" id="L1447">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_8()
		throws Exception {
<span class="nc" id="L1459">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L1460">		Item item = ItemFactory.createItem();</span>
<span class="nc" id="L1461">		int wx = 1;</span>
<span class="nc" id="L1462">		int wy = 1;</span>
<span class="nc" id="L1463">		int wz = 1;</span>

<span class="nc" id="L1465">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
		//       at game.Creature.throwItem(Creature.java:507)
<span class="nc" id="L1473">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_9()
		throws Exception {
<span class="nc" id="L1485">		Creature fixture = CreatureFactory.createCreature2();</span>
<span class="nc" id="L1486">		Item item = ItemFactory.createItem2();</span>
<span class="nc" id="L1487">		int wx = 1;</span>
<span class="nc" id="L1488">		int wy = 1;</span>
<span class="nc" id="L1489">		int wz = 1;</span>

<span class="nc" id="L1491">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
		//       at game.Creature.throwItem(Creature.java:507)
<span class="nc" id="L1499">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_10()
		throws Exception {
<span class="nc" id="L1511">		Creature fixture = CreatureFactory.createCreature3();</span>
<span class="nc" id="L1512">		Item item = ItemFactory.createItem3();</span>
<span class="nc" id="L1513">		int wx = 1;</span>
<span class="nc" id="L1514">		int wy = 1;</span>
<span class="nc" id="L1515">		int wz = 1;</span>

<span class="nc" id="L1517">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.creature(Creature.java:360)
		//       at game.Creature.throwItem(Creature.java:507)
<span class="nc" id="L1525">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_11()
		throws Exception {
<span class="nc" id="L1537">		Creature fixture = CreatureFactory.createCreature4();</span>
<span class="nc" id="L1538">		Item item = ItemFactory.createItem4();</span>
<span class="nc" id="L1539">		int wx = 1;</span>
<span class="nc" id="L1540">		int wy = 1;</span>
<span class="nc" id="L1541">		int wz = 1;</span>

<span class="nc" id="L1543">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
		//       at game.Creature.throwItem(Creature.java:499)
<span class="nc" id="L1550">	}</span>

	/**
	 * Run the void throwItem(Item,int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testThrowItem_12()
		throws Exception {
<span class="nc" id="L1562">		Creature fixture = CreatureFactory.createCreature5();</span>
<span class="nc" id="L1563">		Item item = ItemFactory.createItem();</span>
<span class="nc" id="L1564">		int wx = 1;</span>
<span class="nc" id="L1565">		int wy = 1;</span>
<span class="nc" id="L1566">		int wz = 1;</span>

<span class="nc" id="L1568">		fixture.throwItem(item, wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.realTile(Creature.java:349)
		//       at game.Creature.throwItem(Creature.java:499)
<span class="nc" id="L1575">	}</span>

	/**
	 * Run the Tile tile(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testTile_1()
		throws Exception {
<span class="nc" id="L1587">		Creature fixture = CreatureFactory.createCreature7();</span>
<span class="nc" id="L1588">		int wx = 1;</span>
<span class="nc" id="L1589">		int wy = 1;</span>
<span class="nc" id="L1590">		int wz = 1;</span>

<span class="nc" id="L1592">		Tile result = fixture.tile(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.tile(Creature.java:353)
<span class="nc" id="L1599">		assertNotNull(result);</span>
<span class="nc" id="L1600">	}</span>

	/**
	 * Run the Tile tile(int,int,int) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testTile_2()
		throws Exception {
<span class="nc" id="L1612">		Creature fixture = CreatureFactory.createCreature();</span>
<span class="nc" id="L1613">		int wx = 1;</span>
<span class="nc" id="L1614">		int wy = 1;</span>
<span class="nc" id="L1615">		int wz = 1;</span>

<span class="nc" id="L1617">		Tile result = fixture.tile(wx, wy, wz);</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.canSee(Creature.java:345)
		//       at game.Creature.tile(Creature.java:353)
<span class="nc" id="L1624">		assertNotNull(result);</span>
<span class="nc" id="L1625">	}</span>

	/**
	 * Run the void update() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testUpdate_1()
		throws Exception {
<span class="nc" id="L1637">		Creature fixture = CreatureFactory.createCreature5();</span>

<span class="nc" id="L1639">		fixture.update();</span>

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this test:
		//    java.lang.NullPointerException
		//       at game.Creature.update(Creature.java:240)
<span class="nc" id="L1645">	}</span>

	/**
	 * Run the int visionRadius() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testVisionRadius_1()
		throws Exception {
<span class="nc" id="L1657">		Creature fixture = CreatureFactory.createCreature7();</span>

<span class="nc" id="L1659">		int result = fixture.visionRadius();</span>

		// add additional test code here
<span class="nc" id="L1662">		assertEquals(9, result);</span>
<span class="nc" id="L1663">	}</span>

	/**
	 * Run the Item weapon() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testWeapon_1()
		throws Exception {
<span class="nc" id="L1675">		Creature fixture = CreatureFactory.createCreature5();</span>

<span class="nc" id="L1677">		Item result = fixture.weapon();</span>

		// add additional test code here
<span class="nc" id="L1680">		assertEquals(null, result);</span>
<span class="nc" id="L1681">	}</span>

	/**
	 * Run the int xp() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Test
	public void testXp_1()
		throws Exception {
<span class="nc" id="L1693">		Creature fixture = CreatureFactory.createCreature7();</span>

<span class="nc" id="L1695">		int result = fixture.xp();</span>

		// add additional test code here
<span class="nc" id="L1698">		assertEquals(0, result);</span>
<span class="nc" id="L1699">	}</span>

	/**
	 * Perform pre-test initialization.
	 *
	 * @throws Exception
	 *         if the initialization fails for some reason
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@Before
	public void setUp()
		throws Exception {
		// add additional set up code here
<span class="nc" id="L1713">	}</span>

	/**
	 * Perform post-test clean-up.
	 *
	 * @throws Exception
	 *         if the clean-up fails for some reason
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	@After
	public void tearDown()
		throws Exception {
		// Add additional tear down code here
<span class="nc" id="L1727">	}</span>

	/**
	 * Launch the test.
	 *
	 * @param args the command line arguments
	 *
	 * @generatedBy CodePro at 5/19/18 11:03 AM
	 */
	public static void main(String[] args) {
<span class="nc" id="L1737">		new org.junit.runner.JUnitCore().run(CreatureTest.class);</span>
<span class="nc" id="L1738">	}</span>
}
</pre><div class="footer"><span class="right">Created with <a href="http://www.jacoco.org/jacoco">JaCoCo</a> 0.7.9.201702052155</span>AllUiTests (May 19, 2018 11:46:29 AM)</div></body></html>